### DISCUSSION TITLE
support for PyG (PyTorch Geometric) libraries

### DISCUSSION THREAD

**Discussion by mwzum** [ROCm/ROCm#3655] [Ideas]
Dear ROCm Team,

I am inquiring if there are any plans to support the PyG (PyTorch Geometric) libraries, specifically torch-scatter, torch-sparse, torch-cluster, and torch-spline-conv, on the ROCm platform. The HPC facilities I have access to consist solely of MI250X accelerators, and the lack of support for these libraries is significantly impacting my research.

Thank you for your assistance.

**Comment by fluidnumericsJoe** [comment]
Are you using ORNL, CSC, or Pawsey systems?

**Reply by mwzum** [reply]
Hi Joe,
Its Adastra/CINES

**Reply by fluidnumericsJoe** [reply]
Do you know if they have a program to help allocation awardees port applications to AMD GPUs? Pawsey did something like this with their PACeR program. If not, have your reached out to your help desk / user support services to see if they'd be able to assist with porting ?

I also think probing AMD for more information about the roadmap is a good move - giving a clear plan for the future hasn't really been there strong suit in the software side of things.

**Comment by zichguan-amd** [comment]
Hi @mwzum, PyG support is currently not on our roadmap, but it may be added at some point in the future. In the meantime, external wheels mentioned at https://github.com/pyg-team/pytorch_geometric?tab=readme-ov-file#rocm-wheels are available here [Looong01/pyg-rocm-build](https://github.com/Looong01/pyg-rocm-build) thanks to @Looong01.
